easybuild/tools/github.py

```diff
@@ -835,6 +877,147 @@ def is_patch_for(patch_name, ec):
+                res = not_eligible(msg_tmpl % 'FAILED')
+                test_report_found = True
+            else:
+                print_warning("Failed to determine outcome of test report for comment:\n%s" % comment)
+
+        if 'lgtm' in comment:
+            found_lgtm = True
+
+    if not test_report_found:
+        res = not_eligible(msg_tmpl % "(no test reports found)")
+
+    # check for approved review
+    approved_review_by = []
+    for review in pr_data['reviews']:
+        if review['state'] == 'APPROVED':
+            approved_review_by.append(review['user']['login'])
+
+    msg_tmpl = "* approved review: %s"
+    if approved_review_by:
+        print_msg(msg_tmpl % 'OK (by %s)' % ', '.join(approved_review_by), prefix=False)
+    else:
+        res = not_eligible(msg_tmpl % 'MISSING')
+
+    # check whether a milestone is set
+    msg_tmpl = "* milestone is set: %s"
+    if pr_data['milestone']:
+        print_msg(msg_tmpl % "OK (%s)" % pr_data['milestone']['title'], prefix=False)
+    else:
+        res = not_eligible(msg_tmpl % 'no milestone found')
+
+    return res
+
+
+def merge_pr(pr):
+    """
+    Merge specified pull request
+    """
+    github_user = build_option('github_user')
+    if github_user is None:
+        raise EasyBuildError("GitHub user must be specified to use --merge-pr")
+
+    pr_target_account = build_option('pr_target_account')
+    pr_target_repo = build_option('pr_target_repo')
+
+    pr_url = lambda g: g.repos[pr_target_account][pr_target_repo].pulls[pr]
+    status, pr_data = github_api_get_request(pr_url, github_user)
+    if not status == HTTP_STATUS_OK:
+        raise EasyBuildError("Failed to get data for PR #%d from %s/%s (status: %d %s)",
+                             pr, pr_target_account, pr_target_repo, status, pr_data)
+
+    msg = "\n%s/%s PR #%s was submitted by %s, " % (pr_target_account, pr_target_repo, pr, pr_data['user']['login'])
+    msg += "you are using GitHub account '%s'\n" % github_user
+    print_msg(msg, prefix=False)
+    if pr_data['user']['login'] == github_user:
+        raise EasyBuildError("Please do not merge your own PRs!")
+
+    # also fetch status of last commit
+    pr_head_sha = pr_data['head']['sha']
+    status_url = lambda g: g.repos[pr_target_account][pr_target_repo].commits[pr_head_sha].status
+    status, status_data = github_api_get_request(status_url, github_user)
+    pr_data['status_last_commit'] = status_data['state']
+
+    # also fetch comments
+    comments_url = lambda g: g.repos[pr_target_account][pr_target_repo].issues[pr].comments
+    status, comments_data = github_api_get_request(comments_url, github_user)
+    pr_data['issue_comments'] = comments_data
+
+    # also fetch reviews
+    reviews_url = lambda g: g.repos[pr_target_account][pr_target_repo].pulls[pr].reviews
+    status, reviews_data = github_api_get_request(reviews_url, github_user)
+    pr_data['reviews'] = reviews_data
+
+    force = build_option('force')
```

Comment:
I disagree here, in rare cases it makes sense to merge a PR that doesn't strictly meet all requirements.

For example, when only trivial changes are made that do not add new easyconfigs and do not alter the semantics of existing easyconfigs, we can merge it without requiring a test report. One recent example is https://github.com/easybuilders/easybuild-easyconfigs/pull/4815...

I would rather have maintainers use `--merge-pr X --force` after running it without `--force` first, than favouring manual merges where it's easier to overlook one of the other requirements. Of course, people shouldn't be using `--force` all the time...